In [107]:
import pandas as pd
pd.set_option('display.max_columns', None)
pd.set_option('display.expand_frame_repr', False)
pd.set_option('max_colwidth', None)

In [108]:
df = pd.read_csv('res-stable.csv').query('params.str.len() > 30', engine='python')
df.shape

(8695, 7)

In [111]:
df.tail(20).query('recall > 0.9')

data size           algo  buildtime   querytime                                                                                                                                                params   recall
8687  pca96v2  10M  Learned-index        0.0  656.377887                                                      learned-index-pca96v2-10M-ep=50-lr=0.1-cat=120-model=MLP-buck=4-16706178.meta-pbs.metacentrum.cz  0.91169
8689  pca32v2  10M  Learned-index        0.0  785.125666  learned-index-pca32v2-10M-ep=300-lr=0.01-cat=150-model=MLP-4-prep=True-858417.elixir-pbs.elixir-czech.cz.pkl-buck=7-16720638.meta-pbs.metacentrum.cz  0.91718
8692  pca32v2  10M  Learned-index        0.0  679.323754  learned-index-pca32v2-10M-ep=300-lr=0.01-cat=150-model=MLP-4-prep=True-858417.elixir-pbs.elixir-czech.cz.pkl-buck=6-16720638.meta-pbs.metacentrum.cz  0.90540
8693  pca32v2  10M  Learned-index        0.0  809.390071  learned-index-pca32v2-10M-ep=200-lr=0.01-cat=100-model=MLP-4-prep=True-858408.elixir-pbs.elixir-czech.cz.pkl-buck=6-16720638.meta-pbs.metacentrum.cz  0.93103

In [106]:
df[[0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12]] = df.params.str.split('-', expand=True)
df = df.dropna()
df['ep'] = df[4].str.split('=', expand=True)[1].astype(int)
df['lr'] = df[5].str.split('=', expand=True)[1].astype(float)
df['cat'] = df[6].str.split('=', expand=True)[1].astype(int)
df['model'] = df[8]#.str.split('=', expand=True)[1]
df['bucket'] = df[9].str.split('=', expand=True)[1].astype(int)
df = df.drop([0, 1, 2, 3, 9, 8, 6, 5, 4, 11, 12], axis=1)

ValueError: Columns must be same length as key

In [112]:
df.query('recall > 0.9').sort_values(by='querytime').head(10)

data  size           algo     buildtime   querytime                                                                                                                                                params   recall
8672  pca32v2  100K  Learned-index     10.355728   44.719295                                                   learned-index-pca32v2-100K-ep=10-lr=0.01-cat=20-model=MLP-7-buck=4-16693084.meta-pbs.metacentrum.cz  0.94234
2000  pca96v2  100K  Learned-index     16.622261   70.840965                                                  learned-index-pca96v2-100K-ep=10-lr=0.001-cat=120-model=MLP-buck=6-858044.elixir-pbs.elixir-czech.cz  0.94762
1996  pca96v2  100K  Learned-index     16.622261   85.997628                                                 learned-index-pca96v2-100K-ep=10-lr=0.001-cat=120-model=MLP-buck=12-858044.elixir-pbs.elixir-czech.cz  0.98323
2006  pca96v2  100K  Learned-index     17.047479   98.198620                                                  learned-index-pca96v2-100K-ep=10-lr=0.001-cat=120-model=MLP-buck=6-858040.elixir-pbs.elixir-czech.cz  0.93313
2003  pca96v2  100K  Learned-index     16.622261  104.359989                                                 learned-index-pca96v2-100K-ep=10-lr=0.001-cat=120-model=MLP-buck=24-858044.elixir-pbs.elixir-czech.cz  0.97858
8687  pca96v2   10M  Learned-index      0.000000  656.377887                                                      learned-index-pca96v2-10M-ep=50-lr=0.1-cat=120-model=MLP-buck=4-16706178.meta-pbs.metacentrum.cz  0.91169
478   pca96v2   10M  Learned-index  20604.750362  663.859384                                                 learned-index-pca96v2-10M-ep=200-lr=0.01-cat=120-model=MLP-4-buck=4-858549.elixir-pbs.elixir-czech.cz  0.91421
1399  pca96v2   10M  Learned-index  28435.792981  664.912783                                                  learned-index-pca96v2-10M-ep=210-lr=0.1-cat=110-model=MLP-2-buck=5-858694.elixir-pbs.elixir-czech.cz  0.90162
106   pca96v2   10M  Learned-index  28664.306167  665.981397                                                 learned-index-pca96v2-10M-ep=200-lr=0.01-cat=115-model=MLP-4-buck=4-858548.elixir-pbs.elixir-czech.cz  0.91119
8692  pca32v2   10M  Learned-index      0.000000  679.323754  learned-index-pca32v2-10M-ep=300-lr=0.01-cat=150-model=MLP-4-prep=True-858417.elixir-pbs.elixir-czech.cz.pkl-buck=6-16720638.meta-pbs.metacentrum.cz  0.90540

In [99]:
df.sort_values(by=10).tail(5)

data size           algo     buildtime   querytime                                                                                                params   recall          7             10   ep   lr  cat model  bucket
7734  pca32v2  10M  Learned-index  24598.197472  772.146350  learned-index-pca32v2-10M-ep=200-lr=0.1-cat=115-model=MLP-6-buck=4-858923.elixir-pbs.elixir-czech.cz  0.85803  model=MLP  858923.elixir  200  0.1  115     6       4
6031  pca32v2  10M  Learned-index  25028.898599  953.164181  learned-index-pca32v2-10M-ep=210-lr=0.1-cat=100-model=MLP-6-buck=5-858927.elixir-pbs.elixir-czech.cz  0.90096  model=MLP  858927.elixir  210  0.1  100     6       5
4828  pca32v2  10M  Learned-index  25028.898599  783.370100  learned-index-pca32v2-10M-ep=210-lr=0.1-cat=100-model=MLP-6-buck=4-858927.elixir-pbs.elixir-czech.cz  0.87641  model=MLP  858927.elixir  210  0.1  100     6       4
7373  pca32v2  10M  Learned-index  25177.311780  917.902643  learned-index-pca32v2-10M-ep=210-lr=0.1-cat=110-model=MLP-6-buck=5-858928.elixir-pbs.elixir-czech.cz  0.88535  model=MLP  858928.elixir  210  0.1  110     6       5
4987  pca32v2  10M  Learned-index  25177.311780  754.091750  learned-index-pca32v2-10M-ep=210-lr=0.1-cat=110-model=MLP-6-buck=4-858928.elixir-pbs.elixir-czech.cz  0.86036  model=MLP  858928.elixir  210  0.1  110     6       4

In [66]:
df[10] = df[10].astype(str)

In [93]:
df[df[10] == "858406.elixir"]

data size           algo     buildtime    querytime                                                                                                  params   recall          7             10   ep    lr  cat model  bucket
4006  pca32v2  10M  Learned-index  25246.314968   820.391420   learned-index-pca32v2-10M-ep=180-lr=0.01-cat=200-model=MLP-4-buck=8-858406.elixir-pbs.elixir-czech.cz  0.90965  model=MLP  858406.elixir  180  0.01  200     4       8
4180  pca32v2  10M  Learned-index  25246.314968  1226.541410  learned-index-pca32v2-10M-ep=180-lr=0.01-cat=200-model=MLP-4-buck=12-858406.elixir-pbs.elixir-czech.cz  0.91538  model=MLP  858406.elixir  180  0.01  200     4      12
4644  pca32v2  10M  Learned-index  25246.314968  1019.145122  learned-index-pca32v2-10M-ep=180-lr=0.01-cat=200-model=MLP-4-buck=10-858406.elixir-pbs.elixir-czech.cz  0.91557  model=MLP  858406.elixir  180  0.01  200     4      10

In [94]:
df.query('cat == 200 & recall > 0.9 & bucket <= 12').sort_values(by='querytime')

data size           algo     buildtime    querytime                                                                                                    params   recall          7             10   ep    lr  cat model  bucket
4006    pca32v2  10M  Learned-index  25246.314968   820.391420     learned-index-pca32v2-10M-ep=180-lr=0.01-cat=200-model=MLP-4-buck=8-858406.elixir-pbs.elixir-czech.cz  0.90965  model=MLP  858406.elixir  180  0.01  200     4       8
6581    pca32v2  10M  Learned-index  29538.455934   858.754868     learned-index-pca32v2-10M-ep=200-lr=0.01-cat=200-model=MLP-4-buck=8-858410.elixir-pbs.elixir-czech.cz  0.90734  model=MLP  858410.elixir  200  0.01  200     4       8
7149    pca32v2  10M  Learned-index  43921.529861   872.170779     learned-index-pca32v2-10M-ep=300-lr=0.01-cat=200-model=MLP-4-buck=8-858418.elixir-pbs.elixir-czech.cz  0.90671  model=MLP  858418.elixir  300  0.01  200     4       8
4412    pca32v2  10M  Learned-index  39705.951710   952.793902     learned-index-pca32v2-10M-ep=250-lr=0.01-cat=200-model=MLP-4-buck=8-858414.elixir-pbs.elixir-czech.cz  0.90914  model=MLP  858414.elixir  250  0.01  200     4       8
4644    pca32v2  10M  Learned-index  25246.314968  1019.145122    learned-index-pca32v2-10M-ep=180-lr=0.01-cat=200-model=MLP-4-buck=10-858406.elixir-pbs.elixir-czech.cz  0.91557  model=MLP  858406.elixir  180  0.01  200     4      10
...         ...  ...            ...           ...          ...                                                                                                       ...      ...        ...            ...  ...   ...  ...   ...     ...
2536  clip768v2  10M  Learned-index  15773.666379  3118.639179   learned-index-clip768v2-10M-ep=80-lr=0.01-cat=200-model=MLP-4-buck=12-858505.elixir-pbs.elixir-czech.cz  0.93379  model=MLP  858505.elixir   80  0.01  200     4      12
3294  clip768v2  10M  Learned-index  17738.684199  3227.756413  learned-index-clip768v2-10M-ep=100-lr=0.01-cat=200-model=MLP-4-buck=12-858509.elixir-pbs.elixir-czech.cz  0.92715  model=MLP  858509.elixir  100  0.01  200     4      12
3749  clip768v2  10M  Learned-index  27808.891184  3246.777904  learned-index-clip768v2-10M-ep=160-lr=0.01-cat=200-model=MLP-4-buck=12-858513.elixir-pbs.elixir-czech.cz  0.93516  model=MLP  858513.elixir  160  0.01  200     4      12
2151  clip768v2  10M  Learned-index  15773.666379  5883.633160    learned-index-clip768v2-10M-ep=80-lr=0.01-cat=200-model=MLP-4-buck=8-858505.elixir-pbs.elixir-czech.cz  0.92560  model=MLP  858505.elixir   80  0.01  200     4       8
2660  clip768v2  10M  Learned-index  15773.666379  6466.971434   learned-index-clip768v2-10M-ep=80-lr=0.01-cat=200-model=MLP-4-buck=10-858505.elixir-pbs.elixir-czech.cz  0.93220  model=MLP  858505.elixir   80  0.01  200     4      10

[80 rows x 14 columns]

In [91]:
df.query('cat == 100 & bucket == 4 & recall > 0.9 & data == "pca96v2"').sort_values(by='querytime')

data size           algo     buildtime    querytime                                                                                                  params   recall          7             10   ep     lr  cat model  bucket
765  pca96v2  10M  Learned-index  20772.483462   727.801202   learned-index-pca96v2-10M-ep=200-lr=0.01-cat=100-model=MLP-4-buck=4-858546.elixir-pbs.elixir-czech.cz  0.91924  model=MLP  858546.elixir  200  0.010  100     4       4
91   pca96v2  10M  Learned-index  30584.464664   754.399688   learned-index-pca96v2-10M-ep=210-lr=0.01-cat=100-model=MLP-4-buck=4-858552.elixir-pbs.elixir-czech.cz  0.92055  model=MLP  858552.elixir  210  0.010  100     4       4
791  pca96v2  10M  Learned-index  37775.160467   762.470995   learned-index-pca96v2-10M-ep=300-lr=0.01-cat=100-model=MLP-4-buck=4-858398.elixir-pbs.elixir-czech.cz  0.92281  model=MLP  858398.elixir  300  0.010  100     4       4
66   pca96v2  10M  Learned-index  21357.917504   767.557415   learned-index-pca96v2-10M-ep=180-lr=0.01-cat=100-model=MLP-4-buck=4-858386.elixir-pbs.elixir-czech.cz  0.92047  model=MLP  858386.elixir  180  0.010  100     4       4
614  pca96v2  10M  Learned-index  29139.368009   770.079233  learned-index-pca96v2-10M-ep=200-lr=0.001-cat=100-model=MLP-2-buck=4-858597.elixir-pbs.elixir-czech.cz  0.92393  model=MLP  858597.elixir  200  0.001  100     2       4
..       ...  ...            ...           ...          ...                                                                                                     ...      ...        ...            ...  ...    ...  ...   ...     ...
357  pca96v2  10M  Learned-index  29291.912090  1248.184717  learned-index-pca96v2-10M-ep=200-lr=0.005-cat=100-model=MLP-3-buck=4-853856.elixir-pbs.elixir-czech.cz  0.92144  model=MLP  853856.elixir  200  0.005  100     3       4
302  pca96v2  10M  Learned-index   6237.102150  1293.222758    learned-index-pca96v2-10M-ep=40-lr=0.05-cat=100-model=MLP-7-buck=4-854220.elixir-pbs.elixir-czech.cz  0.91381  model=MLP  854220.elixir   40  0.050  100     7       4
912  pca96v2  10M  Learned-index  44962.648135  1318.502596   learned-index-pca96v2-10M-ep=200-lr=0.01-cat=100-model=MLP-4-buck=4-854132.elixir-pbs.elixir-czech.cz  0.91964  model=MLP  854132.elixir  200  0.010  100     4       4
195  pca96v2  10M  Learned-index   6355.449452  1375.260246    learned-index-pca96v2-10M-ep=40-lr=0.05-cat=100-model=MLP-6-buck=4-853956.elixir-pbs.elixir-czech.cz  0.91381  model=MLP  853956.elixir   40  0.050  100     6       4
505  pca96v2  10M  Learned-index  30041.488086  1993.935694  learned-index-pca96v2-10M-ep=200-lr=0.001-cat=100-model=MLP-8-buck=4-858633.elixir-pbs.elixir-czech.cz  0.92393  model=MLP  858633.elixir  200  0.001  100     8       4

[92 rows x 14 columns]

In [85]:
df.query('cat == 100 & bucket == 4 & recall > 0.9').sort_values(by='querytime')

data size           algo     buildtime    querytime                                                                                                   params   recall          7             10   ep     lr  cat model  bucket
765     pca96v2  10M  Learned-index  20772.483462   727.801202    learned-index-pca96v2-10M-ep=200-lr=0.01-cat=100-model=MLP-4-buck=4-858546.elixir-pbs.elixir-czech.cz  0.91924  model=MLP  858546.elixir  200  0.010  100     4       4
91      pca96v2  10M  Learned-index  30584.464664   754.399688    learned-index-pca96v2-10M-ep=210-lr=0.01-cat=100-model=MLP-4-buck=4-858552.elixir-pbs.elixir-czech.cz  0.92055  model=MLP  858552.elixir  210  0.010  100     4       4
791     pca96v2  10M  Learned-index  37775.160467   762.470995    learned-index-pca96v2-10M-ep=300-lr=0.01-cat=100-model=MLP-4-buck=4-858398.elixir-pbs.elixir-czech.cz  0.92281  model=MLP  858398.elixir  300  0.010  100     4       4
66      pca96v2  10M  Learned-index  21357.917504   767.557415    learned-index-pca96v2-10M-ep=180-lr=0.01-cat=100-model=MLP-4-buck=4-858386.elixir-pbs.elixir-czech.cz  0.92047  model=MLP  858386.elixir  180  0.010  100     4       4
614     pca96v2  10M  Learned-index  29139.368009   770.079233   learned-index-pca96v2-10M-ep=200-lr=0.001-cat=100-model=MLP-2-buck=4-858597.elixir-pbs.elixir-czech.cz  0.92393  model=MLP  858597.elixir  200  0.001  100     2       4
...         ...  ...            ...           ...          ...                                                                                                      ...      ...        ...            ...  ...    ...  ...   ...     ...
2382  clip768v2  10M  Learned-index  15727.658639  2210.267045   learned-index-clip768v2-10M-ep=80-lr=0.01-cat=100-model=MLP-7-buck=4-853225.elixir-pbs.elixir-czech.cz  0.91943  model=MLP  853225.elixir   80  0.010  100     7       4
3727  clip768v2  10M  Learned-index  13546.388715  2277.011052    learned-index-clip768v2-10M-ep=50-lr=0.1-cat=100-model=MLP-6-buck=4-850877.elixir-pbs.elixir-czech.cz  0.90017  model=MLP  850877.elixir   50  0.100  100     6       4
1939  clip768v2  10M  Learned-index  16042.375068  2308.960012   learned-index-clip768v2-10M-ep=80-lr=0.01-cat=100-model=MLP-4-buck=4-858503.elixir-pbs.elixir-czech.cz  0.91874  model=MLP  858503.elixir   80  0.010  100     4       4
3278  clip768v2  10M  Learned-index  38869.315630  2646.561662  learned-index-clip768v2-10M-ep=120-lr=0.05-cat=100-model=MLP-4-buck=4-853115.elixir-pbs.elixir-czech.cz  0.92025  model=MLP  853115.elixir  120  0.050  100     4       4
2183  clip768v2  10M  Learned-index  18395.281229  7014.391047  learned-index-clip768v2-10M-ep=100-lr=0.05-cat=100-model=MLP-3-buck=4-853125.elixir-pbs.elixir-czech.cz  0.92037  model=MLP  853125.elixir  100  0.050  100     3       4

[186 rows x 14 columns]

In [75]:
df.query('querytime < 663 & recall > 0.89 & cat == 100')#.sort_values(by=['recall'], ascending=False).head()

data size           algo     buildtime   querytime                                                                                                  params   recall          7             10   ep     lr  cat model  bucket
231   pca96v2  10M  Learned-index  26159.098549  633.246773   learned-index-pca96v2-10M-ep=200-lr=0.01-cat=100-model=MLP-8-buck=3-854120.elixir-pbs.elixir-czech.cz  0.89156  model=MLP  854120.elixir  200  0.010  100     8       3
465   pca96v2  10M  Learned-index  26652.180314  640.704948  learned-index-pca96v2-10M-ep=200-lr=0.005-cat=100-model=MLP-7-buck=3-854084.elixir-pbs.elixir-czech.cz  0.89347  model=MLP  854084.elixir  200  0.005  100     7       3
466   pca96v2  10M  Learned-index  26234.035828  639.079747  learned-index-pca96v2-10M-ep=200-lr=0.005-cat=100-model=MLP-8-buck=3-853832.elixir-pbs.elixir-czech.cz  0.89347  model=MLP  853832.elixir  200  0.005  100     8       3
485   pca96v2  10M  Learned-index  26144.322119  634.682271  learned-index-pca96v2-10M-ep=200-lr=0.005-cat=100-model=MLP-4-buck=3-854060.elixir-pbs.elixir-czech.cz  0.89347  model=MLP  854060.elixir  200  0.005  100     4       3
494   pca96v2  10M  Learned-index  12970.087719  646.499741  learned-index-pca96v2-10M-ep=100-lr=0.005-cat=100-model=MLP-8-buck=3-853828.elixir-pbs.elixir-czech.cz  0.89348  model=MLP  853828.elixir  100  0.005  100     8       3
575   pca96v2  10M  Learned-index  26106.395778  632.469804   learned-index-pca96v2-10M-ep=200-lr=0.01-cat=100-model=MLP-6-buck=3-854108.elixir-pbs.elixir-czech.cz  0.89156  model=MLP  854108.elixir  200  0.010  100     6       3
753   pca96v2  10M  Learned-index  26280.402088  640.846091  learned-index-pca96v2-10M-ep=200-lr=0.005-cat=100-model=MLP-4-buck=3-853844.elixir-pbs.elixir-czech.cz  0.89347  model=MLP  853844.elixir  200  0.005  100     4       3
820   pca96v2  10M  Learned-index  13264.252489  653.994165  learned-index-pca96v2-10M-ep=100-lr=0.005-cat=100-model=MLP-6-buck=3-853816.elixir-pbs.elixir-czech.cz  0.89348  model=MLP  853816.elixir  100  0.005  100     6       3
859   pca96v2  10M  Learned-index  13524.042039  656.596159  learned-index-pca96v2-10M-ep=100-lr=0.005-cat=100-model=MLP-7-buck=3-853864.elixir-pbs.elixir-czech.cz  0.89348  model=MLP  853864.elixir  100  0.005  100     7       3
927   pca96v2  10M  Learned-index  13171.791899  656.608098  learned-index-pca96v2-10M-ep=100-lr=0.005-cat=100-model=MLP-3-buck=3-854068.elixir-pbs.elixir-czech.cz  0.89348  model=MLP  854068.elixir  100  0.005  100     3       3
1013  pca96v2  10M  Learned-index  26158.759789  645.257889   learned-index-pca96v2-10M-ep=200-lr=0.01-cat=100-model=MLP-7-buck=3-854156.elixir-pbs.elixir-czech.cz  0.89156  model=MLP  854156.elixir  200  0.010  100     7       3
1043  pca96v2  10M  Learned-index  26177.831014  638.377595   learned-index-pca96v2-10M-ep=200-lr=0.01-cat=100-model=MLP-8-buck=3-853904.elixir-pbs.elixir-czech.cz  0.89156  model=MLP  853904.elixir  200  0.010  100     8       3
1266  pca96v2  10M  Learned-index  13278.494591  652.605304  learned-index-pca96v2-10M-ep=100-lr=0.005-cat=100-model=MLP-7-buck=3-854080.elixir-pbs.elixir-czech.cz  0.89348  model=MLP  854080.elixir  100  0.005  100     7       3
1773  pca96v2  10M  Learned-index  26063.764565  634.180433   learned-index-pca96v2-10M-ep=200-lr=0.01-cat=100-model=MLP-3-buck=3-854144.elixir-pbs.elixir-czech.cz  0.89156  model=MLP  854144.elixir  200  0.010  100     3       3
1891  pca96v2  10M  Learned-index  26188.321876  639.648751   learned-index-pca96v2-10M-ep=200-lr=0.01-cat=100-model=MLP-6-buck=3-853892.elixir-pbs.elixir-czech.cz  0.89156  model=MLP  853892.elixir  200  0.010  100     6       3
5712  pca32v2  10M  Learned-index  39785.424431  633.564166   learned-index-pca32v2-10M-ep=300-lr=0.01-cat=100-model=MLP-4-buck=4-858416.elixir-pbs.elixir-czech.cz  0.89280  model=MLP  858416.elixir  300  0.010  100     4       4
7962  pca32v2  10M  Learned-index  25846.324724  579.883077   learned-index-pca32v2-10M-ep=200-lr=0

In [45]:
df.query('data == "pca32v2" & cat == 200 & bucket < 8 & recall > 0.85').sort_values(by='querytime')#.head(5)

data size           algo     buildtime    querytime                                                                                                 params   recall          7             10   ep    lr  cat model  bucket
6382  pca32v2  10M  Learned-index  29413.196878   837.189954  learned-index-pca32v2-10M-ep=200-lr=0.01-cat=200-model=MLP-2-buck=6-856796.elixir-pbs.elixir-czech.cz  0.89476  model=MLP  856796.elixir  200  0.01  200     2       6
7085  pca32v2  10M  Learned-index  13512.001260   842.269878  learned-index-pca32v2-10M-ep=100-lr=0.01-cat=200-model=MLP-6-buck=6-856821.elixir-pbs.elixir-czech.cz  0.89720  model=MLP  856821.elixir  100  0.01  200     6       6
5731  pca32v2  10M  Learned-index   1775.361351   847.248687    learned-index-pca32v2-10M-ep=10-lr=0.1-cat=200-model=MLP-6-buck=6-858224.elixir-pbs.elixir-czech.cz  0.88366  model=MLP  858224.elixir   10  0.10  200     6       6
4412  pca32v2  10M  Learned-index  13580.364249   848.061393   learned-index-pca32v2-10M-ep=100-lr=0.1-cat=200-model=MLP-6-buck=6-856303.elixir-pbs.elixir-czech.cz  0.87057  model=MLP  856303.elixir  100  0.10  200     6       6
6464  pca32v2  10M  Learned-index   5468.751642   857.002352    learned-index-pca32v2-10M-ep=40-lr=0.1-cat=200-model=MLP-2-buck=6-856846.elixir-pbs.elixir-czech.cz  0.87172  model=MLP  856846.elixir   40  0.10  200     2       6
4367  pca32v2  10M  Learned-index   5702.620699   857.162838    learned-index-pca32v2-10M-ep=40-lr=0.1-cat=200-model=MLP-6-buck=6-856876.elixir-pbs.elixir-czech.cz  0.87172  model=MLP  856876.elixir   40  0.10  200     6       6
7387  pca32v2  10M  Learned-index   5582.382492   857.895161    learned-index-pca32v2-10M-ep=40-lr=0.1-cat=200-model=MLP-2-buck=6-856268.elixir-pbs.elixir-czech.cz  0.87172  model=MLP  856268.elixir   40  0.10  200     2       6
4685  pca32v2  10M  Learned-index   5362.327825   862.635539    learned-index-pca32v2-10M-ep=40-lr=0.1-cat=200-model=MLP-3-buck=6-856283.elixir-pbs.elixir-czech.cz  0.87172  model=MLP  856283.elixir   40  0.10  200     3       6
5084  pca32v2  10M  Learned-index  13564.555636   866.346531   learned-index-pca32v2-10M-ep=100-lr=0.1-cat=200-model=MLP-2-buck=6-856273.elixir-pbs.elixir-czech.cz  0.87057  model=MLP  856273.elixir  100  0.10  200     2       6
5936  pca32v2  10M  Learned-index   5592.483427   866.580734    learned-index-pca32v2-10M-ep=40-lr=0.1-cat=200-model=MLP-3-buck=6-856861.elixir-pbs.elixir-czech.cz  0.87172  model=MLP  856861.elixir   40  0.10  200     3       6
5829  pca32v2  10M  Learned-index  13498.212981   938.419325  learned-index-pca32v2-10M-ep=100-lr=0.01-cat=200-model=MLP-2-buck=6-856791.elixir-pbs.elixir-czech.cz  0.89720  model=MLP  856791.elixir  100  0.01  200     2       6
6037  pca32v2  10M  Learned-index   1753.348270   953.420892    learned-index-pca32v2-10M-ep=10-lr=0.1-cat=200-model=MLP-2-buck=6-858212.elixir-pbs.elixir-czech.cz  0.88366  model=MLP  858212.elixir   10  0.10  200     2       6
7649  pca32v2  10M  Learned-index  28409.607595   958.812760   learned-index-pca32v2-10M-ep=200-lr=0.1-cat=200-model=MLP-2-buck=6-856278.elixir-pbs.elixir-czech.cz  0.86814  model=MLP  856278.elixir  200  0.10  200     2       6
7663  pca32v2  10M  Learned-index  27380.639587   958.873317  learned-index-pca32v2-10M-ep=200-lr=0.01-cat=200-model=MLP-3-buck=6-856233.elixir-pbs.elixir-czech.cz  0.89476  model=MLP  856233.elixir  200  0.01  200     3       6
6473  pca32v2  10M  Learned-index   5469.728666   963.279869   learned-index-pca32v2-10M-ep=40-lr=0.01-cat=200-model=MLP-6-buck=6-856816.elixir-pbs.elixir-czech.cz  0.89528  model=MLP  856816.elixir   40  0.01  200     6       6
4904  pca32v2  10M  Learned-index   5461.088558   971.425015   learned-index-pca32v2-10M-ep=40-lr=0.01-cat=200-model=MLP-2-buck=6-856786.elixir-pbs.elixir-czech.cz  0.89528  model=MLP  856786.elixir   40  0.01  200     2       6
4088  pca32v2  10M  Learned-index   5465.837327   972.709558   learned-index-pca32v2-10M-ep=40-lr=0.01-cat=200-model

In [38]:
df.query('cat == 200 & bucket > 3').sort_values(by=10, ascending=False).head(5)

data size           algo     buildtime    querytime                                                                                                  params   recall          7             10   ep    lr  cat model  bucket
461  pca96v2  10M  Learned-index  32301.729142  1408.278545   learned-index-pca96v2-10M-ep=160-lr=0.01-cat=200-model=MLP-4-buck=8-858537.elixir-pbs.elixir-czech.cz  0.92325  model=MLP  858537.elixir  160  0.01  200     4       8
375  pca96v2  10M  Learned-index  32301.729142  2101.129065  learned-index-pca96v2-10M-ep=160-lr=0.01-cat=200-model=MLP-4-buck=12-858537.elixir-pbs.elixir-czech.cz  0.92945  model=MLP  858537.elixir  160  0.01  200     4      12
584  pca96v2  10M  Learned-index  32301.729142  1744.111609  learned-index-pca96v2-10M-ep=160-lr=0.01-cat=200-model=MLP-4-buck=10-858537.elixir-pbs.elixir-czech.cz  0.92699  model=MLP  858537.elixir  160  0.01  200     4      10
129  pca96v2  10M  Learned-index  14186.216230  2141.240584  learned-index-pca96v2-10M-ep=100-lr=0.01-cat=200-model=MLP-4-buck=12-858533.elixir-pbs.elixir-czech.cz  0.93045  model=MLP  858533.elixir  100  0.01  200     4      12
516  pca96v2  10M  Learned-index  14186.216230  1454.790883   learned-index-pca96v2-10M-ep=100-lr=0.01-cat=200-model=MLP-4-buck=8-858533.elixir-pbs.elixir-czech.cz  0.92551  model=MLP  858533.elixir  100  0.01  200     4       8

In [17]:
df.query('querytime < 600 & recall > 0.88').sort_values(by=['recall'], ascending=False).head() #head()#.sort_values(by='querytime').head(5)

data size           algo     buildtime   querytime                                                                                                  params   recall          7             10   ep     lr  cat model  bucket
7849  pca32v2  10M  Learned-index  25846.324724  579.883077   learned-index-pca32v2-10M-ep=200-lr=0.01-cat=100-model=MLP-4-buck=4-858408.elixir-pbs.elixir-czech.cz  0.89217  model=MLP  858408.elixir  200  0.010  100     4       4
305   pca96v2  10M  Learned-index  28065.537984  578.722095    learned-index-pca96v2-10M-ep=210-lr=0.1-cat=100-model=MLP-2-buck=4-858693.elixir-pbs.elixir-czech.cz  0.88820  model=MLP  858693.elixir  210  0.100  100     2       4
553   pca96v2  10M  Learned-index  26521.238766  550.442131  learned-index-pca96v2-10M-ep=200-lr=0.005-cat=120-model=MLP-8-buck=3-854049.elixir-pbs.elixir-czech.cz  0.88401  model=MLP  854049.elixir  200  0.005  120     8       3
1226  pca96v2  10M  Learned-index  27680.569006  556.486657  learned-index-pca96v2-10M-ep=200-lr=0.005-cat=120-model=MLP-6-buck=3-853821.elixir-pbs.elixir-czech.cz  0.88401  model=MLP  853821.elixir  200  0.005  120     6       3
1649  pca96v2  10M  Learned-index  27861.590050  556.654681  learned-index-pca96v2-10M-ep=200-lr=0.005-cat=120-model=MLP-4-buck=3-853845.elixir-pbs.elixir-czech.cz  0.88401  model=MLP  853845.elixir  200  0.005  120     4       3

In [20]:
df.query('querytime < 500 & recall > 0.83').sort_values(by=['recall'], ascending=False).head()

data size           algo     buildtime   querytime                                                                                                  params   recall          7             10   ep     lr  cat model  bucket
1255  pca96v2  10M  Learned-index  13264.252489  448.706479  learned-index-pca96v2-10M-ep=100-lr=0.005-cat=100-model=MLP-6-buck=2-853816.elixir-pbs.elixir-czech.cz  0.83785  model=MLP  853816.elixir  100  0.005  100     6       2
145   pca96v2  10M  Learned-index  13171.791899  448.716240  learned-index-pca96v2-10M-ep=100-lr=0.005-cat=100-model=MLP-3-buck=2-854068.elixir-pbs.elixir-czech.cz  0.83785  model=MLP  854068.elixir  100  0.005  100     3       2
1737  pca96v2  10M  Learned-index  13557.716430  465.959160  learned-index-pca96v2-10M-ep=100-lr=0.005-cat=100-model=MLP-4-buck=2-853840.elixir-pbs.elixir-czech.cz  0.83785  model=MLP  853840.elixir  100  0.005  100     4       2
1581  pca96v2  10M  Learned-index  13524.042039  446.896775  learned-index-pca96v2-10M-ep=100-lr=0.005-cat=100-model=MLP-7-buck=2-853864.elixir-pbs.elixir-czech.cz  0.83785  model=MLP  853864.elixir  100  0.005  100     7       2
1568  pca96v2  10M  Learned-index   9947.336686  483.506230  learned-index-pca96v2-10M-ep=100-lr=0.005-cat=100-model=MLP-8-buck=2-854044.elixir-pbs.elixir-czech.cz  0.83785  model=MLP  854044.elixir  100  0.005  100     8       2

In [27]:
df.query('recall > 0.9 & data == "pca96v2" & cat == 200').sort_values(by='querytime').head(5)

data size           algo     buildtime    querytime                                                                                                 params   recall          7             10   ep    lr  cat model  bucket
592   pca96v2  10M  Learned-index  24875.572941  1205.951210  learned-index-pca96v2-10M-ep=200-lr=0.01-cat=200-model=MLP-4-buck=8-858392.elixir-pbs.elixir-czech.cz  0.92537  model=MLP  858392.elixir  200  0.01  200     4       8
1441  pca96v2  10M  Learned-index  30710.534338  1213.987954  learned-index-pca96v2-10M-ep=250-lr=0.01-cat=200-model=MLP-4-buck=8-858396.elixir-pbs.elixir-czech.cz  0.92448  model=MLP  858396.elixir  250  0.01  200     4       8
461   pca96v2  10M  Learned-index  32301.729142  1408.278545  learned-index-pca96v2-10M-ep=160-lr=0.01-cat=200-model=MLP-4-buck=8-858537.elixir-pbs.elixir-czech.cz  0.92325  model=MLP  858537.elixir  160  0.01  200     4       8
516   pca96v2  10M  Learned-index  14186.216230  1454.790883  learned-index-pca96v2-10M-ep=100-lr=0.01-cat=200-model=MLP-4-buck=8-858533.elixir-pbs.elixir-czech.cz  0.92551  model=MLP  858533.elixir  100  0.01  200     4       8
1673  pca96v2  10M  Learned-index  11443.699270  1459.028274   learned-index-pca96v2-10M-ep=80-lr=0.01-cat=200-model=MLP-4-buck=8-858529.elixir-pbs.elixir-czech.cz  0.92573  model=MLP  858529.elixir   80  0.01  200     4       8

In [5]:
df.query('recall > 0.9').sort_values(by='querytime').head(5)

data size           algo     buildtime   querytime                                                                                                 params   recall          7             10   ep    lr  cat model  bucket
429   pca96v2  10M  Learned-index  20604.750362  663.859384  learned-index-pca96v2-10M-ep=200-lr=0.01-cat=120-model=MLP-4-buck=4-858549.elixir-pbs.elixir-czech.cz  0.91421  model=MLP  858549.elixir  200  0.01  120     4       4
1261  pca96v2  10M  Learned-index  28435.792981  664.912783   learned-index-pca96v2-10M-ep=210-lr=0.1-cat=110-model=MLP-2-buck=5-858694.elixir-pbs.elixir-czech.cz  0.90162  model=MLP  858694.elixir  210  0.10  110     2       5
93    pca96v2  10M  Learned-index  28664.306167  665.981397  learned-index-pca96v2-10M-ep=200-lr=0.01-cat=115-model=MLP-4-buck=4-858548.elixir-pbs.elixir-czech.cz  0.91119  model=MLP  858548.elixir  200  0.01  115     4       4
172   pca96v2  10M  Learned-index  20335.715963  687.072040  learned-index-pca96v2-10M-ep=200-lr=0.01-cat=110-model=MLP-4-buck=4-858547.elixir-pbs.elixir-czech.cz  0.91448  model=MLP  858547.elixir  200  0.01  110     4       4
7637  pca32v2  10M  Learned-index  26998.736559  693.058218  learned-index-pca32v2-10M-ep=200-lr=0.01-cat=150-model=MLP-4-buck=6-858409.elixir-pbs.elixir-czech.cz  0.90436  model=MLP  858409.elixir  200  0.01  150     4       6

In [80]:
df.query('recall > 0.9').sort_values(by='querytime').head(1)

data size           algo     buildtime   querytime                                                                                                  params   recall          7             10   ep     lr  cat model  bucket
1287  pca96v2  10M  Learned-index  26521.238766  727.405015  learned-index-pca96v2-10M-ep=200-lr=0.005-cat=120-model=MLP-8-buck=4-854049.elixir-pbs.elixir-czech.cz  0.91463  model=MLP  854049.elixir  200  0.005  120     8       4

In [79]:
df.sort_values(by=[10]).tail(10)

data size           algo     buildtime    querytime                                                                                                 params   recall          7             10   ep     lr  cat model  bucket
4381  pca32v2  10M  Learned-index  12467.625299   574.306925   learned-index-pca32v2-10M-ep=100-lr=0.1-cat=120-model=MLP-6-buck=3-856880.elixir-pbs.elixir-czech.cz  0.82033  model=MLP  856880.elixir  100  0.100  120     6       3
6673  pca32v2  10M  Learned-index  12467.625299   196.431916   learned-index-pca32v2-10M-ep=100-lr=0.1-cat=120-model=MLP-6-buck=1-856880.elixir-pbs.elixir-czech.cz  0.58015  model=MLP  856880.elixir  100  0.100  120     6       1
6525  pca32v2  10M  Learned-index  17078.325650  2404.981756  learned-index-pca32v2-10M-ep=100-lr=0.1-cat=200-model=MLP-6-buck=10-856881.elixir-pbs.elixir-czech.cz  0.91677  model=MLP  856881.elixir  100  0.100  200     6      10
4801  pca32v2  10M  Learned-index  17078.325650  9550.767678  learned-index-pca32v2-10M-ep=100-lr=0.1-cat=200-model=MLP-6-buck=40-856881.elixir-pbs.elixir-czech.cz  0.97765  model=MLP  856881.elixir  100  0.100  200     6      40
5151  pca32v2  10M  Learned-index  17078.325650  1447.287215   learned-index-pca32v2-10M-ep=100-lr=0.1-cat=200-model=MLP-6-buck=6-856881.elixir-pbs.elixir-czech.cz  0.87057  model=MLP  856881.elixir  100  0.100  200     6       6
4446  pca32v2  10M  Learned-index  17078.325650  4766.919472  learned-index-pca32v2-10M-ep=100-lr=0.1-cat=200-model=MLP-6-buck=20-856881.elixir-pbs.elixir-czech.cz  0.95756  model=MLP  856881.elixir  100  0.100  200     6      20
6500  pca32v2  10M  Learned-index  17078.325650   494.090205   learned-index-pca32v2-10M-ep=100-lr=0.1-cat=200-model=MLP-6-buck=2-856881.elixir-pbs.elixir-czech.cz  0.69080  model=MLP  856881.elixir  100  0.100  200     6       2
4868  pca32v2  10M  Learned-index    211.320360  1532.916239   learned-index-pca32v2-10M-ep=1-lr=0.001-cat=200-model=MLP-2-buck=2-858082.elixir-pbs.elixir-czech.cz  0.68402  model=MLP  858082.elixir    1  0.001  200     2       2
7188  pca32v2  10M  Learned-index    180.015363  1628.374989   learned-index-pca32v2-10M-ep=1-lr=0.001-cat=200-model=MLP-6-buck=2-858094.elixir-pbs.elixir-czech.cz  0.68402  model=MLP  858094.elixir    1  0.001  200     6       2
5410  pca32v2  10M  Learned-index    178.862515  1630.156575   learned-index-pca32v2-10M-ep=1-lr=0.001-cat=200-model=MLP-8-buck=2-858101.elixir-pbs.elixir-czech.cz  0.68402  model=MLP  858101.elixir    1  0.001  200     8       2

## Overall top 3

In [54]:
df.sort_values(by='recall', ascending=False).head(2)

data size           algo     buildtime    querytime                                                                                                  params   recall          7             10   ep    lr  cat model  bucket
1241  pca96v2  10M  Learned-index  26579.057275  4314.171264  learned-index-pca96v2-10M-ep=200-lr=0.01-cat=120-model=MLP-8-buck=24-854890.elixir-pbs.elixir-czech.cz  0.98586  model=MLP  854890.elixir  200  0.01  120     8      24
989   pca96v2  10M  Learned-index  26520.581224  4315.988252  learned-index-pca96v2-10M-ep=200-lr=0.01-cat=120-model=MLP-6-buck=24-854875.elixir-pbs.elixir-czech.cz  0.98586  model=MLP  854875.elixir  200  0.01  120     6      24

In [55]:
df.query('recall > 0.9').sort_values(by='querytime').head(5)

data size           algo     buildtime   querytime                                                                                                  params   recall          7             10   ep     lr  cat model  bucket
1287  pca96v2  10M  Learned-index  26521.238766  727.405015  learned-index-pca96v2-10M-ep=200-lr=0.005-cat=120-model=MLP-8-buck=4-854049.elixir-pbs.elixir-czech.cz  0.91463  model=MLP  854049.elixir  200  0.005  120     8       4
856   pca96v2  10M  Learned-index  26166.820423  728.180541   learned-index-pca96v2-10M-ep=200-lr=0.01-cat=120-model=MLP-4-buck=4-853917.elixir-pbs.elixir-czech.cz  0.91480  model=MLP  853917.elixir  200  0.010  120     4       4
1149  pca96v2  10M  Learned-index  26791.926345  730.595510  learned-index-pca96v2-10M-ep=200-lr=0.005-cat=120-model=MLP-7-buck=4-853869.elixir-pbs.elixir-czech.cz  0.91463  model=MLP  853869.elixir  200  0.005  120     7       4
503   pca96v2  10M  Learned-index  26571.924829  731.163253  learned-index-pca96v2-10M-ep=200-lr=0.005-cat=120-model=MLP-4-buck=4-854061.elixir-pbs.elixir-czech.cz  0.91463  model=MLP  854061.elixir  200  0.005  120     4       4
850   pca96v2  10M  Learned-index  26367.848469  732.406396   learned-index-pca96v2-10M-ep=200-lr=0.05-cat=120-model=MLP-6-buck=4-853965.elixir-pbs.elixir-czech.cz  0.90059  model=MLP  853965.elixir  200  0.050  120     6       4

In [73]:
df.query('data == "pca96v2" & ep > 200')#.sort_values(by='querytime').head(10)

Empty DataFrame
Columns: [data, size, algo, buildtime, querytime, params, recall, 7, 10, ep, lr, cat, model, bucket]
Index: []

In [70]:
df.query('data == "pca32v2" & recall > 0.87 & recall < 0.9').sort_values(by='querytime').head(10)

data size           algo    buildtime   querytime                                                                                                params   recall          7             10  ep    lr  cat model  bucket
6497  pca32v2  10M  Learned-index  7557.644634  734.671338   learned-index-pca32v2-10M-ep=60-lr=0.01-cat=80-model=MLP-7-buck=3-852902.elixir-pbs.elixir-czech.cz  0.87039  model=MLP  852902.elixir  60  0.01   80     7       3
4672  pca32v2  10M  Learned-index  7408.173200  734.847385   learned-index-pca32v2-10M-ep=60-lr=0.01-cat=80-model=MLP-4-buck=3-852852.elixir-pbs.elixir-czech.cz  0.87039  model=MLP  852852.elixir  60  0.01   80     4       3
7051  pca32v2  10M  Learned-index  7690.177685  736.174443   learned-index-pca32v2-10M-ep=60-lr=0.01-cat=80-model=MLP-8-buck=3-852827.elixir-pbs.elixir-czech.cz  0.87039  model=MLP  852827.elixir  60  0.01   80     8       3
6449  pca32v2  10M  Learned-index  7984.597668  756.733068  learned-index-pca32v2-10M-ep=60-lr=0.01-cat=120-model=MLP-3-buck=4-852881.elixir-pbs.elixir-czech.cz  0.88192  model=MLP  852881.elixir  60  0.01  120     3       4
5095  pca32v2  10M  Learned-index  7959.096520  757.401146  learned-index-pca32v2-10M-ep=60-lr=0.01-cat=120-model=MLP-7-buck=4-852906.elixir-pbs.elixir-czech.cz  0.88192  model=MLP  852906.elixir  60  0.01  120     7       4
6212  pca32v2  10M  Learned-index  7898.355867  758.968206  learned-index-pca32v2-10M-ep=60-lr=0.05-cat=120-model=MLP-3-buck=4-852731.elixir-pbs.elixir-czech.cz  0.87335  model=MLP  852731.elixir  60  0.05  120     3       4
6555  pca32v2  10M  Learned-index  7807.372833  761.168578  learned-index-pca32v2-10M-ep=60-lr=0.01-cat=120-model=MLP-8-buck=4-852831.elixir-pbs.elixir-czech.cz  0.88192  model=MLP  852831.elixir  60  0.01  120     8       4
4443  pca32v2  10M  Learned-index  7960.923100  764.057818   learned-index-pca32v2-10M-ep=60-lr=0.01-cat=80-model=MLP-7-buck=3-852452.elixir-pbs.elixir-czech.cz  0.87039  model=MLP  852452.elixir  60  0.01   80     7       3
5046  pca32v2  10M  Learned-index  8163.700746  768.628307   learned-index-pca32v2-10M-ep=60-lr=0.01-cat=80-model=MLP-6-buck=3-852802.elixir-pbs.elixir-czech.cz  0.87039  model=MLP  852802.elixir  60  0.01   80     6       3
5893  pca32v2  10M  Learned-index  6348.660966  768.701044   learned-index-pca32v2-10M-ep=50-lr=0.01-cat=80-model=MLP-3-buck=3-852872.elixir-pbs.elixir-czech.cz  0.87166  model=MLP  852872.elixir  50  0.01   80     3       3

In [60]:
df.query('data == "clip768v2" & recall > 0.9').sort_values(by='querytime').head(10)

data size           algo     buildtime   querytime                                                                                                  params   recall          7             10   ep    lr  cat model  bucket
2368  clip768v2  10M  Learned-index  25511.012582  885.140054  learned-index-clip768v2-10M-ep=120-lr=0.01-cat=90-model=MLP-6-buck=3-853444.elixir-pbs.elixir-czech.cz  0.90234  model=MLP  853444.elixir  120  0.01   90     6       3
3120  clip768v2  10M  Learned-index  16651.616141  886.920622   learned-index-clip768v2-10M-ep=80-lr=0.01-cat=90-model=MLP-6-buck=3-853164.elixir-pbs.elixir-czech.cz  0.90039  model=MLP  853164.elixir   80  0.01   90     6       3
1360  clip768v2  10M  Learned-index  25533.967646  888.100942  learned-index-clip768v2-10M-ep=120-lr=0.01-cat=90-model=MLP-6-buck=3-853174.elixir-pbs.elixir-czech.cz  0.90234  model=MLP  853174.elixir  120  0.01   90     6       3
1902  clip768v2  10M  Learned-index  25190.170352  888.820744  learned-index-clip768v2-10M-ep=120-lr=0.01-cat=90-model=MLP-8-buck=3-853189.elixir-pbs.elixir-czech.cz  0.90234  model=MLP  853189.elixir  120  0.01   90     8       3
1529  clip768v2  10M  Learned-index  24964.191060  889.186543  learned-index-clip768v2-10M-ep=120-lr=0.01-cat=90-model=MLP-8-buck=3-853459.elixir-pbs.elixir-czech.cz  0.90234  model=MLP  853459.elixir  120  0.01   90     8       3
2132  clip768v2  10M  Learned-index  16918.425007  889.598526   learned-index-clip768v2-10M-ep=80-lr=0.01-cat=90-model=MLP-4-buck=3-853464.elixir-pbs.elixir-czech.cz  0.90039  model=MLP  853464.elixir   80  0.01   90     4       3
2938  clip768v2  10M  Learned-index  16956.492706  891.850923   learned-index-clip768v2-10M-ep=80-lr=0.01-cat=90-model=MLP-3-buck=3-853479.elixir-pbs.elixir-czech.cz  0.90039  model=MLP  853479.elixir   80  0.01   90     3       3
3072  clip768v2  10M  Learned-index  16828.462366  892.262139   learned-index-clip768v2-10M-ep=80-lr=0.01-cat=90-model=MLP-4-buck=3-853194.elixir-pbs.elixir-czech.cz  0.90039  model=MLP  853194.elixir   80  0.01   90     4       3
1450  clip768v2  10M  Learned-index  16922.874708  892.776631   learned-index-clip768v2-10M-ep=80-lr=0.01-cat=90-model=MLP-8-buck=3-853179.elixir-pbs.elixir-czech.cz  0.90039  model=MLP  853179.elixir   80  0.01   90     8       3
2288  clip768v2  10M  Learned-index  16780.920681  893.555992   learned-index-clip768v2-10M-ep=80-lr=0.01-cat=90-model=MLP-7-buck=3-853224.elixir-pbs.elixir-czech.cz  0.90039  model=MLP  853224.elixir   80  0.01   90     7       3

In [53]:
df.query('data == "pca32v2" & recall > 0.9 & cat < 50').sort_values(by='querytime').head(10)

data size           algo     buildtime    querytime                                                                                                params   recall          7             10   ep    lr  cat model  bucket
6016  pca32v2  10M  Learned-index   4787.345850  1576.220044    learned-index-pca32v2-10M-ep=50-lr=0.1-cat=30-model=MLP-6-buck=3-851055.elixir-pbs.elixir-czech.cz  0.91268  model=MLP  851055.elixir   50  0.10   30     6       3
5643  pca32v2  10M  Learned-index   5897.350472  1628.278539   learned-index-pca32v2-10M-ep=50-lr=0.01-cat=30-model=MLP-8-buck=3-851315.elixir-pbs.elixir-czech.cz  0.92199  model=MLP  851315.elixir   50  0.01   30     8       3
3497  pca32v2  10M  Learned-index   5959.287112  1629.352546   learned-index-pca32v2-10M-ep=50-lr=0.01-cat=30-model=MLP-4-buck=3-851335.elixir-pbs.elixir-czech.cz  0.92199  model=MLP  851335.elixir   50  0.01   30     4       3
6093  pca32v2  10M  Learned-index   5940.658014  1634.955842   learned-index-pca32v2-10M-ep=50-lr=0.01-cat=30-model=MLP-6-buck=3-851295.elixir-pbs.elixir-czech.cz  0.92199  model=MLP  851295.elixir   50  0.01   30     6       3
6109  pca32v2  10M  Learned-index   2372.085245  1637.069026    learned-index-pca32v2-10M-ep=20-lr=0.1-cat=30-model=MLP-4-buck=3-851090.elixir-pbs.elixir-czech.cz  0.91325  model=MLP  851090.elixir   20  0.10   30     4       3
5900  pca32v2  10M  Learned-index   5938.499697  1640.298772   learned-index-pca32v2-10M-ep=50-lr=0.01-cat=30-model=MLP-4-buck=3-851155.elixir-pbs.elixir-czech.cz  0.92199  model=MLP  851155.elixir   50  0.01   30     4       3
3420  pca32v2  10M  Learned-index  11723.173602  1642.866582  learned-index-pca32v2-10M-ep=100-lr=0.01-cat=30-model=MLP-4-buck=3-851340.elixir-pbs.elixir-czech.cz  0.92147  model=MLP  851340.elixir  100  0.01   30     4       3
4569  pca32v2  10M  Learned-index   6212.808212  1643.628891   learned-index-pca32v2-10M-ep=50-lr=0.01-cat=30-model=MLP-8-buck=3-851135.elixir-pbs.elixir-czech.cz  0.92199  model=MLP  851135.elixir   50  0.01   30     8       3
6138  pca32v2  10M  Learned-index   2346.796720  1645.024266    learned-index-pca32v2-10M-ep=20-lr=0.1-cat=30-model=MLP-8-buck=3-851250.elixir-pbs.elixir-czech.cz  0.91325  model=MLP  851250.elixir   20  0.10   30     8       3
6980  pca32v2  10M  Learned-index   2285.401348  1646.010887    learned-index-pca32v2-10M-ep=20-lr=0.1-cat=30-model=MLP-8-buck=3-851070.elixir-pbs.elixir-czech.cz  0.91325  model=MLP  851070.elixir   20  0.10   30     8       3

In [47]:
df.query('data == "pca32v2" & recall > 0.9 & bucket <= 10').sort_values(by='querytime').head(10)

data size           algo    buildtime    querytime                                                                                               params   recall          7             10  ep    lr  cat model  bucket
4046  pca32v2  10M  Learned-index  7690.177685   966.491667  learned-index-pca32v2-10M-ep=60-lr=0.01-cat=80-model=MLP-8-buck=4-852827.elixir-pbs.elixir-czech.cz  0.90669  model=MLP  852827.elixir  60  0.01   80     8       4
7269  pca32v2  10M  Learned-index  7557.644634   970.329134  learned-index-pca32v2-10M-ep=60-lr=0.01-cat=80-model=MLP-7-buck=4-852902.elixir-pbs.elixir-czech.cz  0.90669  model=MLP  852902.elixir  60  0.01   80     7       4
6121  pca32v2  10M  Learned-index  7408.173200   970.918052  learned-index-pca32v2-10M-ep=60-lr=0.01-cat=80-model=MLP-4-buck=4-852852.elixir-pbs.elixir-czech.cz  0.90669  model=MLP  852852.elixir  60  0.01   80     4       4
5504  pca32v2  10M  Learned-index  5026.206680   995.982049  learned-index-pca32v2-10M-ep=40-lr=0.01-cat=80-model=MLP-8-buck=4-852817.elixir-pbs.elixir-czech.cz  0.90313  model=MLP  852817.elixir  40  0.01   80     8       4
4275  pca32v2  10M  Learned-index  5071.450600  1002.094921  learned-index-pca32v2-10M-ep=40-lr=0.01-cat=80-model=MLP-7-buck=4-852892.elixir-pbs.elixir-czech.cz  0.90313  model=MLP  852892.elixir  40  0.01   80     7       4
5891  pca32v2  10M  Learned-index  6348.660966  1006.278766  learned-index-pca32v2-10M-ep=50-lr=0.01-cat=80-model=MLP-3-buck=4-852872.elixir-pbs.elixir-czech.cz  0.90524  model=MLP  852872.elixir  50  0.01   80     3       4
5617  pca32v2  10M  Learned-index  8163.700746  1009.303083  learned-index-pca32v2-10M-ep=60-lr=0.01-cat=80-model=MLP-6-buck=4-852802.elixir-pbs.elixir-czech.cz  0.90669  model=MLP  852802.elixir  60  0.01   80     6       4
6315  pca32v2  10M  Learned-index  5087.815979  1013.601717  learned-index-pca32v2-10M-ep=40-lr=0.01-cat=80-model=MLP-4-buck=4-852842.elixir-pbs.elixir-czech.cz  0.90313  model=MLP  852842.elixir  40  0.01   80     4       4
4777  pca32v2  10M  Learned-index  7878.805274  1017.268376  learned-index-pca32v2-10M-ep=60-lr=0.01-cat=80-model=MLP-6-buck=4-852352.elixir-pbs.elixir-czech.cz  0.90669  model=MLP  852352.elixir  60  0.01   80     6       4
5084  pca32v2  10M  Learned-index  5215.284468  1018.233810  learned-index-pca32v2-10M-ep=40-lr=0.01-cat=80-model=MLP-7-buck=4-852442.elixir-pbs.elixir-czech.cz  0.90313  model=MLP  852442.elixir  40  0.01   80     7       4

In [35]:
df.query('data == "pca32v2" & recall > 0.9').sort_values(by='querytime')

data size           algo     buildtime     querytime                                                                                                  params   recall          7             10   ep    lr  cat model  bucket
4046  pca32v2  10M  Learned-index   7690.177685    966.491667     learned-index-pca32v2-10M-ep=60-lr=0.01-cat=80-model=MLP-8-buck=4-852827.elixir-pbs.elixir-czech.cz  0.90669  model=MLP  852827.elixir   60  0.01   80     8       4
7269  pca32v2  10M  Learned-index   7557.644634    970.329134     learned-index-pca32v2-10M-ep=60-lr=0.01-cat=80-model=MLP-7-buck=4-852902.elixir-pbs.elixir-czech.cz  0.90669  model=MLP  852902.elixir   60  0.01   80     7       4
6121  pca32v2  10M  Learned-index   7408.173200    970.918052     learned-index-pca32v2-10M-ep=60-lr=0.01-cat=80-model=MLP-4-buck=4-852852.elixir-pbs.elixir-czech.cz  0.90669  model=MLP  852852.elixir   60  0.01   80     4       4
5504  pca32v2  10M  Learned-index   5026.206680    995.982049     learned-index-pca32v2-10M-ep=40-lr=0.01-cat=80-model=MLP-8-buck=4-852817.elixir-pbs.elixir-czech.cz  0.90313  model=MLP  852817.elixir   40  0.01   80     8       4
4275  pca32v2  10M  Learned-index   5071.450600   1002.094921     learned-index-pca32v2-10M-ep=40-lr=0.01-cat=80-model=MLP-7-buck=4-852892.elixir-pbs.elixir-czech.cz  0.90313  model=MLP  852892.elixir   40  0.01   80     7       4
...       ...  ...            ...           ...           ...                                                                                                     ...      ...        ...            ...  ...   ...  ...   ...     ...
6249  pca32v2  10M  Learned-index   5943.812714   9171.161808      learned-index-pca32v2-10M-ep=50-lr=0.1-cat=10-model=MLP-8-buck=2-851074.elixir-pbs.elixir-czech.cz  0.91303  model=MLP  851074.elixir   50   0.1   10     8       2
5485  pca32v2  10M  Learned-index  12920.532139   9259.608832     learned-index-pca32v2-10M-ep=100-lr=0.1-cat=10-model=MLP-8-buck=2-851079.elixir-pbs.elixir-czech.cz  0.91610  model=MLP  851079.elixir  100   0.1   10     8       2
4798  pca32v2  10M  Learned-index  17078.325650   9550.767678   learned-index-pca32v2-10M-ep=100-lr=0.1-cat=200-model=MLP-6-buck=40-856881.elixir-pbs.elixir-czech.cz  0.97765  model=MLP  856881.elixir  100   0.1  200     6      40
6735  pca32v2  10M  Learned-index   6493.359762  10717.308626     learned-index-pca32v2-10M-ep=50-lr=0.01-cat=10-model=MLP-8-buck=3-851134.elixir-pbs.elixir-czech.cz  0.96038  model=MLP  851134.elixir   50  0.01   10     8       3
5798  pca32v2  10M  Learned-index  31804.070995  11342.831924  learned-index-pca32v2-10M-ep=200-lr=0.01-cat=120-model=MLP-3-buck=24-856810.elixir-pbs.elixir-czech.cz  0.98197  model=MLP  856810.elixir  200  0.01  120     3      24

[720 rows x 14 columns]

In [ ]:
df.query('recall > 0.9').sort_values(by='querytime').head(5)

In [13]:
def remove_file(f):
    import os
    os.remove(f"result/pca32v2/10M/{f}.h5")
    
for f in df.sort_values(by=10).tail(49).params.values:
    remove_file(f)

In [119]:
df.sort_values(by=10).tail(5)

data size           algo     buildtime    querytime                                                                                                params   recall          7             10   ep   lr  cat model  bucket
5733  pca32v2  10M  Learned-index   5275.091199   832.939928   learned-index-pca32v2-10M-ep=40-lr=0.1-cat=120-model=MLP-6-buck=3-856297.elixir-pbs.elixir-czech.cz  0.81178  model=MLP  856297.elixir   40  0.1  120     6       3
6333  pca32v2  10M  Learned-index   5275.091199  3198.973144  learned-index-pca32v2-10M-ep=40-lr=0.1-cat=120-model=MLP-6-buck=12-856297.elixir-pbs.elixir-czech.cz  0.95099  model=MLP  856297.elixir   40  0.1  120     6      12
3893  pca32v2  10M  Learned-index   5151.456435  1304.141337   learned-index-pca32v2-10M-ep=40-lr=0.1-cat=200-model=MLP-6-buck=6-856298.elixir-pbs.elixir-czech.cz  0.87176  model=MLP  856298.elixir   40  0.1  200     6       6
5816  pca32v2  10M  Learned-index   5151.456435   445.137952   learned-index-pca32v2-10M-ep=40-lr=0.1-cat=200-model=MLP-6-buck=2-856298.elixir-pbs.elixir-czech.cz  0.69063  model=MLP  856298.elixir   40  0.1  200     6       2
3406  pca32v2  10M  Learned-index  12915.253560   197.130571  learned-index-pca32v2-10M-ep=100-lr=0.1-cat=120-model=MLP-6-buck=1-856302.elixir-pbs.elixir-czech.cz  0.58015  model=MLP  856302.elixir  100  0.1  120     6       1

## Search time -- 6 buckets
[2023-07-13 10:49:43,863][INFO ][__main__] Inference time: 0.12289190292358398
[2023-07-13 10:49:43,863][INFO ][__main__] Search time: 1304.1413373947144
[2023-07-13 10:49:43,863][INFO ][__main__] Search single time: 213.12875747680664 (* 6) = 1278
[2023-07-13 10:49:43,863][INFO ][__main__] Sequential search time: 166.14448833465576 (* 6) = 996

#### Top 1 by data

In [11]:
df.query('recall > 0.9').groupby('data')['querytime'].min()#query('recall > 0.9').sort_values(by='querytime').head(3)

data
clip768v2    853.549118
pca32v2      966.491667
pca96v2      727.405015
Name: querytime, dtype: float64

#### Top 1 by bucket size

In [12]:
df.query('recall > 0.9').groupby('bucket')['querytime'].min()

bucket
2     1587.177841
3      853.549118
4      727.405015
6     1077.392715
10    1411.663729
12    2115.920972
20    2772.028177
24    4191.389168
40    5428.713855
Name: querytime, dtype: float64

In [122]:
df.query('bucket > 5').sort_values(by='querytime').head(3)

data size           algo     buildtime   querytime                                                                                                params   recall          7             10   ep   lr  cat model  bucket
6427  pca32v2  10M  Learned-index   5582.382492  857.895161   learned-index-pca32v2-10M-ep=40-lr=0.1-cat=200-model=MLP-2-buck=6-856268.elixir-pbs.elixir-czech.cz  0.87172  model=MLP  856268.elixir   40  0.1  200     2       6
4006  pca32v2  10M  Learned-index   5362.327825  862.635539   learned-index-pca32v2-10M-ep=40-lr=0.1-cat=200-model=MLP-3-buck=6-856283.elixir-pbs.elixir-czech.cz  0.87172  model=MLP  856283.elixir   40  0.1  200     3       6
4362  pca32v2  10M  Learned-index  13564.555636  866.346531  learned-index-pca32v2-10M-ep=100-lr=0.1-cat=200-model=MLP-2-buck=6-856273.elixir-pbs.elixir-czech.cz  0.87057  model=MLP  856273.elixir  100  0.1  200     2       6

#### Top 1 by n. of categories

In [103]:
df.query('recall > 0.9').groupby('cat')['querytime'].min()

cat
10     2807.695384
20     1587.177841
30     1576.220044
50     1047.837158
80      926.336974
90      877.095950
100     836.385029
110    1020.847102
120     727.405015
Name: querytime, dtype: float64

In [10]:
df.sort_values(by=['querytime'])

data  size           algo     buildtime     querytime                                                                                             params   recall
1525  pca32v2    100K  Learned-index  39.764608     5.016663      learned-index-pca32v2-100K-ep=30-lr=0.5-cat=50-model=MLP-buck=1-843249.elixir-pbs.elixir-czech.cz  0.46584
1524  pca32v2    100K  Learned-index  39.673587     8.504480      learned-index-pca32v2-100K-ep=30-lr=0.5-cat=50-model=MLP-buck=2-843265.elixir-pbs.elixir-czech.cz  0.62520
1526  pca32v2    100K  Learned-index  39.673587     12.072524     learned-index-pca32v2-100K-ep=30-lr=0.5-cat=50-model=MLP-buck=3-843265.elixir-pbs.elixir-czech.cz  0.68785
1523  pca32v2    100K  Learned-index  113.795209    18.563406     learned-index                                                                                      0.90973
1527  pca32v2    300K  Learned-index  348.149078    32.989279     learned-index                                                                                      0.87796
...       ...     ...            ...         ...          ...               ...                                                                                          ...
59    clip768v2  10M   Learned-index  20656.892781  22630.376532  learned-index-clip768v2-emb-10M-100-0.05-50                                                        0.97896
170   clip768v2  10M   Learned-index  10600.792844  38683.646897  learned-index-clip768v2-emb-10M-50-0.05-100                                                        0.96822
368   clip768v2  10M   Learned-index  10682.141847  39344.094956  learned-index-clip768v2-emb-10M-50-0.1-100                                                         0.95461
404   clip768v2  10M   Learned-index  10247.390891  43583.894748  learned-index-clip768v2-emb-10M-50-0.1-50                                                          0.98800
198   clip768v2  10M   Learned-index  10384.350073  44233.590862  learned-index-clip768v2-emb-10M-50-0.05-50                                                         0.98928

[1528 rows x 7 columns]

In [4]:
df.query('querytime < 1100 & recall > 0.9').sort_values(by='recall', ascending=False).head(30)

data size           algo     buildtime    querytime                                                                                                   params   recall
2048  clip768v2  10M  Learned-index  21016.299465  1099.494444  learned-index-clip768v2-10M-ep=100-lr=0.01-cat=100-model=MLP-8-buck=4-850962.elixir-pbs.elixir-czech.cz  0.92321
2301  clip768v2  10M  Learned-index  25928.267981  995.499643   learned-index-clip768v2-10M-ep=120-lr=0.01-cat=120-model=MLP-7-buck=4-853237.elixir-pbs.elixir-czech.cz  0.92276
1773  clip768v2  10M  Learned-index  25701.544988  1007.527989  learned-index-clip768v2-10M-ep=120-lr=0.01-cat=120-model=MLP-3-buck=4-853492.elixir-pbs.elixir-czech.cz  0.92276
974   clip768v2  10M  Learned-index  25715.498433  995.077134   learned-index-clip768v2-10M-ep=120-lr=0.01-cat=120-model=MLP-6-buck=4-853177.elixir-pbs.elixir-czech.cz  0.92276
2442  clip768v2  10M  Learned-index  25252.616198  993.405510   learned-index-clip768v2-10M-ep=120-lr=0.01-cat=120-model=MLP-8-buck=4-853462.elixir-pbs.elixir-czech.cz  0.92276
2427  clip768v2  10M  Learned-index  26376.203247  1027.578987  learned-index-clip768v2-10M-ep=120-lr=0.01-cat=120-model=MLP-buck=4-853432.elixir-pbs.elixir-czech.cz    0.92276
1286  clip768v2  10M  Learned-index  25759.758898  992.142598   learned-index-clip768v2-10M-ep=120-lr=0.01-cat=120-model=MLP-4-buck=4-853207.elixir-pbs.elixir-czech.cz  0.92276
1866  clip768v2  10M  Learned-index  25491.309324  997.950636   learned-index-clip768v2-10M-ep=120-lr=0.01-cat=120-model=MLP-buck=4-853162.elixir-pbs.elixir-czech.cz    0.92276
2089  clip768v2  10M  Learned-index  26114.155997  1020.847102  learned-index-clip768v2-10M-ep=120-lr=0.01-cat=110-model=MLP-8-buck=4-853191.elixir-pbs.elixir-czech.cz  0.92264
1880  clip768v2  10M  Learned-index  25548.482920  1026.395593  learned-index-clip768v2-10M-ep=120-lr=0.01-cat=110-model=MLP-buck=4-853161.elixir-pbs.elixir-czech.cz    0.92264
1958  clip768v2  10M  Learned-index  25809.628520  1029.094863  learned-index-clip768v2-10M-ep=120-lr=0.01-cat=110-model=MLP-4-buck=4-853206.elixir-pbs.elixir-czech.cz  0.92264
2325  clip768v2  10M  Learned-index  25227.156776  1027.171227  learned-index-clip768v2-10M-ep=120-lr=0.01-cat=110-model=MLP-7-buck=4-853236.elixir-pbs.elixir-czech.cz  0.92264
2373  clip768v2  10M  Learned-index  25627.606670  1021.466440  learned-index-clip768v2-10M-ep=120-lr=0.01-cat=110-model=MLP-7-buck=4-853506.elixir-pbs.elixir-czech.cz  0.92264
1662  clip768v2  10M  Learned-index  25534.350258  1030.081169  learned-index-clip768v2-10M-ep=120-lr=0.01-cat=110-model=MLP-3-buck=4-853491.elixir-pbs.elixir-czech.cz  0.92264
2516  clip768v2  10M  Learned-index  25650.722640  1028.908049  learned-index-clip768v2-10M-ep=120-lr=0.01-cat=110-model=MLP-buck=4-853431.elixir-pbs.elixir-czech.cz    0.92264
2151  clip768v2  10M  Learned-index  25663.937609  1021.004459  learned-index-clip768v2-10M-ep=120-lr=0.01-cat=110-model=MLP-6-buck=4-853446.elixir-pbs.elixir-czech.cz  0.92264
470   pca96v2    10M  Learned-index  13171.791899  853.966211   learned-index-pca96v2-10M-ep=100-lr=0.005-cat=100-model=MLP-3-buck=4-854068.elixir-pbs.elixir-czech.cz   0.92171
338   pca96v2    10M  Learned-index  13524.042039  860.291974   learned-index-pca96v2-10M-ep=100-lr=0.005-cat=100-model=MLP-7-buck=4-853864.elixir-pbs.elixir-czech.cz   0.92171
618   pca96v2    10M  Learned-index  13264.252489  853.170330   learned-index-pca96v2-10M-ep=100-lr=0.005-cat=100-model=MLP-6-buck=4-853816.elixir-pbs.elixir-czech.cz   0.92171
576   pca96v2    10M  Learned-index  12970.087719  851.316769   learned-index-pca96v2-10M-ep=100-lr=0.005-cat=100-model=MLP-8-buck=4-853828.elixir-pbs.elixir-czech.cz   0.92171
536   pca96v2    10M  Learned-index  9947.336686   935.879552   learned-index-pca96v2-10M-ep=100-lr=0.005-cat=100-model=MLP-8-buck=4-854044.elixir-pbs.elixir-czech.cz   0.92171
524   pca96v2    10M  Learned-index  13557.716430  881.197219   learned-index-pca96v2-10M-ep=100-lr=0.005-cat=100-mod

In [7]:
# python search/search.py --size=100K --n-categories=2000 --epochs=80 --model=MLP-7 --preprocess=True --lr=0.01 --save=True

df.query('querytime < 1100 & recall > 0.9').sort_values(by='recall', ascending=False).head(50)

data  size           algo     buildtime    querytime                                                                                                   params   recall
447   clip768v2  10M   Learned-index  21016.299465  1099.494444  learned-index-clip768v2-10M-ep=100-lr=0.01-cat=100-model=MLP-8-buck=4-850962.elixir-pbs.elixir-czech.cz  0.92321
2416  pca32v2    100K  Learned-index  113.795209    18.563406    learned-index                                                                                            0.90973
711   pca32v2    10M   Learned-index  10163.213147  1049.239424  learned-index-pca32v2-10M-ep=80-lr=0.005-cat=80-model=MLP-3-buck=4-852127.elixir-pbs.elixir-czech.cz     0.90741
853   pca32v2    10M   Learned-index  10502.029406  1044.571484  learned-index-pca32v2-10M-ep=80-lr=0.005-cat=80-model=MLP-8-buck=4-852096.elixir-pbs.elixir-czech.cz     0.90741
1986  pca32v2    10M   Learned-index  10255.796429  1048.048363  learned-index-pca32v2-10M-ep=80-lr=0.005-cat=80-model=MLP-7-buck=4-852142.elixir-pbs.elixir-czech.cz     0.90741
1994  pca32v2    10M   Learned-index  10210.993582  1052.415275  learned-index-pca32v2-10M-ep=80-lr=0.005-cat=80-model=MLP-4-buck=4-852111.elixir-pbs.elixir-czech.cz     0.90741
2015  pca32v2    10M   Learned-index  10532.473953  1043.639671  learned-index-pca32v2-10M-ep=80-lr=0.005-cat=80-model=MLP-6-buck=4-852081.elixir-pbs.elixir-czech.cz     0.90741
2280  pca32v2    10M   Learned-index  8155.963824   1068.089615  learned-index-pca32v2-10M-ep=80-lr=0.005-cat=80-model=MLP-8-buck=4-851871.elixir-pbs.elixir-czech.cz     0.90741
2360  pca32v2    10M   Learned-index  8055.153278   1059.191891  learned-index-pca32v2-10M-ep=80-lr=0.005-cat=80-model=MLP-4-buck=4-851886.elixir-pbs.elixir-czech.cz     0.90741
826   pca32v2    10M   Learned-index  10156.629495  1042.946798  learned-index-pca32v2-10M-ep=100-lr=0.005-cat=80-model=MLP-8-buck=4-851876.elixir-pbs.elixir-czech.cz    0.90733
955   pca32v2    10M   Learned-index  10095.914603  1039.351954  learned-index-pca32v2-10M-ep=80-lr=0.01-cat=80-model=MLP-7-buck=4-852066.elixir-pbs.elixir-czech.cz      0.90487
1571  pca32v2    10M   Learned-index  10258.852425  1041.895324  learned-index-pca32v2-10M-ep=80-lr=0.01-cat=80-model=MLP-8-buck=4-852021.elixir-pbs.elixir-czech.cz      0.90487
1867  pca32v2    10M   Learned-index  10097.786843  1040.407610  learned-index-pca32v2-10M-ep=80-lr=0.01-cat=80-model=MLP-3-buck=4-852051.elixir-pbs.elixir-czech.cz      0.90487
1886  pca32v2    10M   Learned-index  10080.440187  1045.898922  learned-index-pca32v2-10M-ep=80-lr=0.01-cat=80-model=MLP-4-buck=4-852036.elixir-pbs.elixir-czech.cz      0.90487
2231  pca32v2    10M   Learned-index  10010.834522  1044.695525  learned-index-pca32v2-10M-ep=80-lr=0.01-cat=80-model=MLP-6-buck=4-852006.elixir-pbs.elixir-czech.cz      0.90487
701   pca32v2    10M   Learned-index  5161.166816   1047.731406  learned-index-pca32v2-10M-ep=40-lr=0.01-cat=80-model=MLP-buck=4-852317.elixir-pbs.elixir-czech.cz        0.90313

In [12]:
df.query('querytime < 1200 & recall > 0.9').sort_values(by='recall', ascending=False).head(30)

data  size           algo     buildtime    querytime                                                                                                   params   recall
447   clip768v2  10M   Learned-index  21016.299465  1099.494444  learned-index-clip768v2-10M-ep=100-lr=0.01-cat=100-model=MLP-8-buck=4-850962.elixir-pbs.elixir-czech.cz  0.92321
460   clip768v2  10M   Learned-index  21340.775154  1151.715770  learned-index-clip768v2-10M-ep=100-lr=0.01-cat=100-model=MLP-6-buck=4-850762.elixir-pbs.elixir-czech.cz  0.92321
663   clip768v2  10M   Learned-index  21213.454384  1114.680401  learned-index-clip768v2-10M-ep=100-lr=0.01-cat=100-model=MLP-7-buck=4-850262.elixir-pbs.elixir-czech.cz  0.92321
75    clip768v2  10M   Learned-index  20979.731317  1118.044588  learned-index-clip768v2-10M-ep=100-lr=0.01-cat=100-model=MLP-buck=4-850302.elixir-pbs.elixir-czech.cz    0.92321
125   clip768v2  10M   Learned-index  21357.786327  1105.344402  learned-index-clip768v2-10M-ep=100-lr=0.01-cat=100-model=MLP-6-buck=4-850942.elixir-pbs.elixir-czech.cz  0.92321
637   clip768v2  10M   Learned-index  21180.777470  1104.608187  learned-index-clip768v2-10M-ep=100-lr=0.01-cat=100-model=MLP-3-buck=4-850462.elixir-pbs.elixir-czech.cz  0.92321
493   clip768v2  10M   Learned-index  21326.989718  1113.140971  learned-index-clip768v2-10M-ep=100-lr=0.01-cat=100-model=MLP-8-buck=4-850782.elixir-pbs.elixir-czech.cz  0.92321
467   clip768v2  10M   Learned-index  21290.158103  1112.881765  learned-index-clip768v2-10M-ep=100-lr=0.01-cat=100-model=MLP-4-buck=4-850802.elixir-pbs.elixir-czech.cz  0.92321
1523  pca32v2    100K  Learned-index  113.795209    18.563406    learned-index                                                                                            0.90973
12    clip768v2  10M   Learned-index  10714.075028  1138.870404  learned-index-clip768v2-10M-ep=50-lr=0.1-cat=100-model=MLP-3-buck=4-850217.elixir-pbs.elixir-czech.cz    0.90002
382   clip768v2  10M   Learned-index  10719.116047  1128.842316  learned-index-clip768v2-10M-ep=50-lr=0.1-cat=100-model=MLP-3-buck=4-850397.elixir-pbs.elixir-czech.cz    0.90002
186   clip768v2  10M   Learned-index  10627.008720  1130.061887  learned-index-clip768v2-10M-ep=50-lr=0.1-cat=100-model=MLP-7-buck=4-850377.elixir-pbs.elixir-czech.cz    0.90002
602   clip768v2  10M   Learned-index  10740.272677  1140.427086  learned-index-clip768v2-10M-ep=50-lr=0.1-cat=100-model=MLP-buck=4-850417.elixir-pbs.elixir-czech.cz      0.90002
631   clip768v2  10M   Learned-index  10646.805039  1139.612172  learned-index-clip768v2-10M-ep=50-lr=0.1-cat=100-model=MLP-buck=4-850237.elixir-pbs.elixir-czech.cz      0.90002
177   clip768v2  10M   Learned-index  10784.808275  1138.546153  learned-index-clip768v2-10M-ep=50-lr=0.1-cat=100-model=MLP-8-buck=4-850897.elixir-pbs.elixir-czech.cz    0.90002
35    clip768v2  10M   Learned-index  10705.382592  1146.524477  learned-index-clip768v2-10M-ep=50-lr=0.1-cat=100-model=MLP-7-buck=4-850197.elixir-pbs.elixir-czech.cz    0.90002
8     clip768v2  10M   Learned-index  10516.590594  1140.795134  learned-index-clip768v2-10M-ep=50-lr=0.1-cat=100-model=MLP-4-buck=4-850737.elixir-pbs.elixir-czech.cz    0.90002

In [ ]:
df.query('querytime < 1100 & recall > 0.89').sort_values(by='recall', ascending=False).head(50)

In [4]:
df.query('querytime < 1100 & recall > 0.9').sort_values(by='recall', ascending=False).head(30)

data  size           algo   buildtime  querytime         params   recall
47  pca32v2  100K  Learned-index  113.795209  18.563406  learned-index  0.90973

In [36]:
# python search/search.py --size=10M --n-categories=47 --epochs=27 --model=MLP-7 --preprocess=True --lr=0.025 --save=True
df.query('params.str.startswith("learned-index-pca32v2-10M-ep=27-lr=0.025-cat=47-model=MLP-7-buck=3")', engine='python')

data size           algo    buildtime    querytime                                                                                                params   recall
5413  pca32v2  10M  Learned-index  3347.686355  1219.883782  learned-index-pca32v2-10M-ep=27-lr=0.025-cat=47-model=MLP-7-buck=3-845662.elixir-pbs.elixir-czech.cz  0.90041
5538  pca32v2  10M  Learned-index  2781.950249  1090.942238  learned-index-pca32v2-10M-ep=27-lr=0.025-cat=47-model=MLP-7-buck=3-846743.elixir-pbs.elixir-czech.cz  0.90258
7556  pca32v2  10M  Learned-index  4073.361228  1605.131101  learned-index-pca32v2-10M-ep=27-lr=0.025-cat=47-model=MLP-7-buck=3-849446.elixir-pbs.elixir-czech.cz  0.89838

In [37]:
# best so far -- elmo5-16
df.query('querytime < 1100 & recall > 0.9').sort_values(by='recall', ascending=False).head(30)

data  size           algo     buildtime    querytime                                                                                                params   recall
10477  pca32v2  100K  Learned-index  113.795209    18.563406    learned-index                                                                                         0.90973
8829   pca32v2  10M   Learned-index  14400.000000  1087.746459  learned-index-pca32v2-pca32-10M-50-0.1-100-16619702.meta-pbs.metacentrum.cz                           0.90828
5538   pca32v2  10M   Learned-index  2781.950249   1090.942238  learned-index-pca32v2-10M-ep=27-lr=0.025-cat=47-model=MLP-7-buck=3-846743.elixir-pbs.elixir-czech.cz  0.90258
7970   pca32v2  10M   Learned-index  3375.201787   1095.131285  learned-index-pca32v2-10M-ep=28-lr=0.02-cat=48-model=MLP-8-buck=3-849025.elixir-pbs.elixir-czech.cz   0.90066
3388   pca32v2  10M   Learned-index  3748.134555   1094.398245  learned-index-pca32v2-10M-ep=30-lr=0.025-cat=50-model=MLP-6-buck=3-843901.elixir-pbs.elixir-czech.cz  0.90044
8073   pca32v2  10M   Learned-index  3851.038334   1099.067293  learned-index-pca32v2-10M-ep=32-lr=0.023-cat=52-model=MLP-buck=3-846860.elixir-pbs.elixir-czech.cz    0.90019

In [22]:
# best so far -- elmo5-16
df.query('querytime < 1100 & recall > 0.9').sort_values(by='recall', ascending=False).head(30)

data  size           algo     buildtime    querytime                                                                                                params   recall
10477  pca32v2  100K  Learned-index  113.795209    18.563406    learned-index                                                                                         0.90973
8829   pca32v2  10M   Learned-index  14400.000000  1087.746459  learned-index-pca32v2-pca32-10M-50-0.1-100-16619702.meta-pbs.metacentrum.cz                           0.90828
5538   pca32v2  10M   Learned-index  2781.950249   1090.942238  learned-index-pca32v2-10M-ep=27-lr=0.025-cat=47-model=MLP-7-buck=3-846743.elixir-pbs.elixir-czech.cz  0.90258
7970   pca32v2  10M   Learned-index  3375.201787   1095.131285  learned-index-pca32v2-10M-ep=28-lr=0.02-cat=48-model=MLP-8-buck=3-849025.elixir-pbs.elixir-czech.cz   0.90066
3388   pca32v2  10M   Learned-index  3748.134555   1094.398245  learned-index-pca32v2-10M-ep=30-lr=0.025-cat=50-model=MLP-6-buck=3-843901.elixir-pbs.elixir-czech.cz  0.90044
8073   pca32v2  10M   Learned-index  3851.038334   1099.067293  learned-index-pca32v2-10M-ep=32-lr=0.023-cat=52-model=MLP-buck=3-846860.elixir-pbs.elixir-czech.cz    0.90019

In [5]:
# best so far -- elmo5-16
df.query('querytime < 1100 & recall > 0.9').sort_values(by='recall', ascending=False).head(30)

data  size           algo     buildtime    querytime                                                                                                params   recall
8983  pca32v2  100K  Learned-index  113.795209    18.563406    learned-index                                                                                         0.90973
7568  pca32v2  10M   Learned-index  14400.000000  1087.746459  learned-index-pca32v2-pca32-10M-50-0.1-100-16619702.meta-pbs.metacentrum.cz                           0.90828
4735  pca32v2  10M   Learned-index  2781.950249   1090.942238  learned-index-pca32v2-10M-ep=27-lr=0.025-cat=47-model=MLP-7-buck=3-846743.elixir-pbs.elixir-czech.cz  0.90258
2898  pca32v2  10M   Learned-index  3748.134555   1094.398245  learned-index-pca32v2-10M-ep=30-lr=0.025-cat=50-model=MLP-6-buck=3-843901.elixir-pbs.elixir-czech.cz  0.90044
6925  pca32v2  10M   Learned-index  3851.038334   1099.067293  learned-index-pca32v2-10M-ep=32-lr=0.023-cat=52-model=MLP-buck=3-846860.elixir-pbs.elixir-czech.cz    0.90019

In [25]:
# best so far -- elwe5
df.query('querytime < 1200 & recall > 0.9').sort_values(by='recall', ascending=False).head(30)

data  size           algo     buildtime    querytime                                                                                                params   recall
4026  pca32v2  100K  Learned-index  113.795209    18.563406    learned-index                                                                                         0.90973
3375  pca32v2  10M   Learned-index  14400.000000  1087.746459  learned-index-pca32v2-pca32-10M-50-0.1-100-16619702.meta-pbs.metacentrum.cz                           0.90828
3226  pca32v2  10M   Learned-index  3575.913363   1189.828057  learned-index-pca32v2-10M-ep=29-lr=0.027-cat=48-model=MLP-buck=3-845319.elixir-pbs.elixir-czech.cz    0.90185
2247  pca32v2  10M   Learned-index  3091.070361   1166.253631  learned-index-pca32v2-10M-ep=25-lr=0.027-cat=52-model=MLP-6-buck=3-845363.elixir-pbs.elixir-czech.cz  0.90130
742   pca32v2  10M   Learned-index  3729.609501   1131.147942  learned-index-pca32v2-10M-ep=30-lr=0.025-cat=46-model=MLP-buck=3-845541.elixir-pbs.elixir-czech.cz    0.90129
2858  pca32v2  10M   Learned-index  3080.658372   1180.062169  learned-index-pca32v2-10M-ep=25-lr=0.027-cat=48-model=MLP-7-buck=3-845431.elixir-pbs.elixir-czech.cz  0.90114
1256  pca32v2  10M   Learned-index  3776.701467   1153.905299  learned-index-pca32v2-10M-ep=30-lr=0.02-cat=45-model=MLP-buck=3-844655.elixir-pbs.elixir-czech.cz     0.90108
2591  pca32v2  10M   Learned-index  3365.684825   1196.609176  learned-index-pca32v2-10M-ep=27-lr=0.027-cat=48-model=MLP-6-buck=3-845375.elixir-pbs.elixir-czech.cz  0.90047
1334  pca32v2  10M   Learned-index  3748.134555   1094.398245  learned-index-pca32v2-10M-ep=30-lr=0.025-cat=50-model=MLP-6-buck=3-843901.elixir-pbs.elixir-czech.cz  0.90044
2574  pca32v2  10M   Learned-index  3794.191259   1109.310425  learned-index-pca32v2-10M-ep=30-lr=0.025-cat=47-model=MLP-buck=3-845542.elixir-pbs.elixir-czech.cz    0.90036
950   pca32v2  10M   Learned-index  3668.261391   1107.164901  learned-index-pca32v2-10M-ep=29-lr=0.025-cat=47-model=MLP-buck=3-845534.elixir-pbs.elixir-czech.cz    0.90015

In [12]:
df.sort_values(by='recall', ascending=False).head(15)

data  size           algo     buildtime     querytime  \
2    clip768v2   10M  Learned-index  10384.350073  44233.590862   
5    clip768v2   10M  Learned-index  10247.390891  43583.894748   
0    clip768v2   10M  Learned-index  20656.892781  22630.376532   
122    pca32v2   10M  Learned-index  12259.877640  40856.643734   
91     pca32v2   10M  Learned-index   6656.743498  43646.492634   
3    clip768v2   10M  Learned-index  20725.218157  22569.086937   
74     pca32v2   10M  Learned-index   6586.360964  45811.651632   
1    clip768v2   10M  Learned-index  10600.792844  38683.646897   
4    clip768v2   10M  Learned-index  10682.141847  39344.094956   
105    pca32v2   10M  Learned-index  14400.000000   1372.487562   
152    pca32v2  100K  Learned-index    113.795209     18.563406   
127    pca32v2   10M  Learned-index  14400.000000   1087.746459   
86     pca32v2   10M  Learned-index   3583.866385   1078.301981   
6    clip768v2  100K  Learned-index    104.615377    380.361706   
14     pca32v2   10M  Learned-index   3730.611985   1119.872534   

                                                params   recall  
2           learned-index-clip768v2-emb-10M-50-0.05-50  0.98928  
5            learned-index-clip768v2-emb-10M-50-0.1-50  0.98800  
0          learned-index-clip768v2-emb-10M-100-0.05-50  0.97896  
122        learned-index-pca32v2-pca32-10M-100-0.05-50  0.97438  
91          learned-index-pca32v2-pca32-10M-50-0.05-50  0.97340  
3           learned-index-clip768v2-emb-10M-100-0.1-50  0.97254  
74           learned-index-pca32v2-pca32-10M-50-0.1-50  0.96987  
1          learned-index-clip768v2-emb-10M-50-0.05-100  0.96822  
4           learned-index-clip768v2-emb-10M-50-0.1-100  0.95461  
105  learned-index-pca32v2-pca32-10M-50-0.1-100-166...  0.92514  
152                                      learned-index  0.90973  
127  learned-index-pca32v2-pca32-10M-50-0.1-100-166...  0.90828  
86   learned-index-pca32v2-10M-ep=30-lr=0.05-cat=50...  0.89652  
6          learned-index-clip768v2-emb-100K-50-0.1-100  0.89616  
14   learned-index-pca32v2-10M-ep=30-lr=0.05-cat=50...  0.89470

In [14]:
df.loc[127].params

'learned-index-pca32v2-pca32-10M-50-0.1-100-16619702.meta-pbs.metacentrum.cz'